### PR TITLE
Simplify PartitionArrangement

### PR DESCRIPTION
**Pull request by viliam-durina**

### CODE REVIEW

**Comment by hz-highfive** [comment]
The job `jet-oss-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/jet-oss-pr-builder/3028/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/jet-oss-pr-builder/3028/jet-oss-pr-builder-3028.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>

</details>
</pre>

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/execution/init/PartitionArrangement.java [hunk 1]
```diff
@@ -35,62 +33,58 @@
...
+    Map<Address, int[]> getRemotePartitionAssignment() {
+        return remotePartitionAssignment;
     }
 
     /**
-     * Determines for each processor instance the partition IDs it will be in charge of
-     * (processors are identified by their index). The method is called separately for
-     * each edge. For a distributed edge, only partitions owned by the local member need
-     * to be assigned; for a non-distributed edge, every partition ID must be assigned.
-     * Local partitions will get the same assignments in both cases, and repeating the
```
Comment (cangencer) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/execution/init/PartitionArrangement.java:54]: are we removing this condition? if so, why?

Comment (viliam-durina) [reply hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/execution/init/PartitionArrangement.java:54]: Yes, this PR is removing it? I'd reverse the question: why would we keep it? It's not used, it's not documented and it's very rarely beneficial. So the reason is to simplify the code.

Comment (cangencer) [reply hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/execution/init/PartitionArrangement.java:54]: As discussed, currently we don't make use of this feature anywhere and it makes the code unnecessarily complex.

**Comment by viliam-durina** [comment]
verify

**Pull request merged**